### PR TITLE
Forked and Modified NR Log Ingestion Lambda to use image_uri option as well

### PR DESCRIPTION
**Pull request by shourabh-bh**
- The NR Log Ingestion terraform code is managed by NR team and is not-compatible (uses only local filesystem) with BH infrastructure. So the only way out is to fork it and make the relevant change to make the code compatible for all infra.

- So in this PR I have forked the NR managed repo and made the relevant changes so that it can use either "image_uri" or "local filename" as an option.

JIRA :: https://blinkhealth.atlassian.net/browse/INFRAENG-3645

### CODE REVIEW

**Review by sachinchemmala — COMMENTED** [review]
1 - Can you please pull out all the variables to a separate file?
2 - Please add a sample TF plan output or link a PR which has it.

Other than that LGTM

**Comment by shourabh-bh** [comment]
> 1 - Can you please pull out all the variables to a separate file? 2 - Please add a sample TF plan output or link a PR that has it.
> 
> Other than that LGTM

1. @sachinchemmala This module is a NR managed module and I don't want to make any structural changes there. I am just doing the minimal changes and raised [this](https://github.com/newrelic/aws-log-ingestion/pull/92) PR for NR team to approve.

2. This module is referenced in this [PR](https://github.com/blinkhealth/terraform-lambda/pull/53) . We can see the TF plan in the mentioned PR

File: terraform.tf [hunk 1]
```diff
@@ -197,6 +209,10 @@ resource "aws_lambda_function" "ingestion_function" {
     }
   }
 
+  lifecycle {
```
Comment (sachinchemmala) [inline terraform.tf:212]: Can't we make this optional as we discussed?

Comment (shourabh-bh) [reply terraform.tf:212]: @sachinchemmala I tried to make it option but it did not worked. This block doesn't support variables.

Comment (sachinchemmala) [reply terraform.tf:212]: Have you tried the dynamic block? I remember doing similar with dynamic block

Comment (shourabh-bh) [reply terraform.tf:212]: Let me check if dynamic block is supported in this case or not ?

**Review by sachinchemmala — APPROVED** [review]
LGTM. please add a comment why we need lifecycle rule before merging

File: terraform.tf (hunk 1 above)
Comment (shourabh-bh) [reply terraform.tf:212]: I tried and came to a conclusion that dynamic will also not work as `lifecycle` is an Explicit dependency which won't support variables as input

**Comment by shourabh-bh** [comment]
> LGTM. please add a comment why we need lifecycle rule before merging

The first time deployment of lambda will happen using terraform code. But next time when a new image tag is pushed in the ECR then the lambda image_uri will be updated using [this](https://github.com/blinkhealth/terraform-lambda/pull/53/files#diff-9eda380c7c316c7ffafec06a9d8530afe8dc13b0204c5141ad1d9289292979e4R83) GA Workflow. So I want the image_uri option in the TF code should be ignored so that if in future ever this TF code runs, my existing lambda should start pointing to the old tag

**Pull request merged**